src/Hl7.Fhir.Support.Poco/Specification/Source/IResourceResolver.cs

```diff
@@ -0,0 +1,53 @@
+﻿/* 
+ * Copyright (c) 2016, Firely (info@fire.ly) and contributors
+ * See the file CONTRIBUTORS for details.
+ * 
+ * This file is licensed under the BSD 3-Clause license
+ * available at https://raw.githubusercontent.com/FirelyTeam/fhir-net-api/master/LICENSE
+ */
+
+using System;
+using System.Threading.Tasks;
+using Hl7.Fhir.Model;
+
+namespace Hl7.Fhir.Specification.Source
+{
+    /// <summary>Interface for resolving FHIR artifacts by (canonical) uri.</summary>
+#pragma warning disable CS0618 // Type or member is obsolete
+    public interface IResourceResolver : ISyncOrAsyncResourceResolver
```

Comment:
Should we not mark this interface as Obsolete?

Reply:
That is a difficult question that I have spent quite some time on. To be perfectly honest, in v2.x, the synchronous interface will still be used in a large part of the codebase. So although we plan to go all-async, this is certainly not yet the case. With this in mind, I decided to NOT make it obsolete yet - since it is in actual use, without any fixed plans to remove it. That's also why I went out of my way to support both sync and async scenarios.

Another very practical argument, that is a consequence of this: Marking this obsolete will generate so many obsolete messages that they become totally useless - you'll just skip looking at them.

Reply:
Maybe `[Obsolete]` on constructor - at least we discourage new use of the old interface?

Reply:
Actually, this is not possible, sorry ;-)   There is only one constructor on e.g. DirectorySource, and it takes ISyncOrAsyncResolver, so it cannot be marked `[Obsolete]`.

Reply:
Ok, then we leave it like this. Thanks for the research.